contracts/interfaces/modules/licensing/IPILicenseTemplate.sol

```diff
@@ -0,0 +1,60 @@
+/// @param mintingFee The fee to be paid when minting a license.
+/// @param expiration The expiration period of the license.
+/// @param commercialUse Indicates whether the work can be used commercially or not.
+/// @param commercialAttribution whether attribution is required when reproducing the work commercially or not.
+/// @param commercializerChecker commercializers that are allowed to commercially exploit the work. If zero address,
+/// then no restrictions is enforced.
+/// @param commercializerCheckerData The data to be passed to the commercializer checker contract.
+/// @param commercialRevShare Percentage of revenue that must be shared with the licensor.
+/// @param commercialRevCelling The maximum revenue that can be generated from the commercial use of the work.
+/// @param derivativesAllowed Indicates whether the licensee can create derivatives of his work or not.
+/// @param derivativesAttribution Indicates whether attribution is required for derivatives of the work or not.
+/// @param derivativesApproval Indicates whether the licensor must approve derivatives of the work before they can be
+/// linked to the licensor IP ID or not.
+/// @param derivativesReciprocal Indicates whether the licensee must license derivatives of the work under the
+/// same terms or not.
+/// @param derivativeRevCelling The maximum revenue that can be generated from the derivative use of the work.
+/// @param currency The ERC20 token to be used to pay the minting fee. the token must be registered in story protocol.
+struct PILTerms {
+    bool transferable;
+    address royaltyPolicy;
+    uint256 mintingFee;
+    uint256 expiration;
+    bool commercialUse;
+    bool commercialAttribution;
+    address commercializerChecker;
+    bytes commercializerCheckerData;
+    uint32 commercialRevShare;
+    uint256 commercialRevCelling;
+    bool derivativesAllowed;
+    bool derivativesAttribution;
+    bool derivativesApproval;
+    bool derivativesReciprocal;
+    uint256 derivativeRevCelling;
+    address currency;
+}
+
+/// @title IPILicenseTemplate
+/// @notice This interface defines the methods for a Programmable IP License (PIL) template.
+/// The PIL template is used to generate PIL terms that can be attached to IP Assets.
+/// The legal document of the PIL can be found in this repository.
+interface IPILicenseTemplate is ILicenseTemplate {
+    /// @notice Registers new license terms.
+    /// @param terms The PILTerms to register.
+    /// @return selectedLicenseTermsId The ID of the newly registered license terms.
+    function registerLicenseTerms(PILTerms calldata terms) external returns (uint256 selectedLicenseTermsId);
+
+    /// @notice Gets the ID of the given license terms.
+    /// @param terms The PILTerms to get the ID for.
+    /// @return selectedLicenseTermsId The ID of the given license terms.
+    function getLicenseTermsId(PILTerms calldata terms) external view returns (uint256 selectedLicenseTermsId);
```

Comment:
same here, can this function be moved to ILicenseTemplate?

Reply:
As the function need the license date type "PILTerms" specific to the PILicenseTemplate, hence need to be placed in the PILicenseTemplate.